dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorVulkan.java

```diff
@@ -0,0 +1,31 @@
+package com.dat3m.dartagnan.program.processing.compilation;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.event.arch.vulkan.VulkanRMW;
+import com.dat3m.dartagnan.program.event.core.Event;
+import com.dat3m.dartagnan.program.event.core.Load;
+import com.dat3m.dartagnan.program.event.core.rmw.RMWStore;
+
+import java.util.List;
+
+import static com.dat3m.dartagnan.program.event.EventFactory.*;
+
+public class VisitorVulkan extends VisitorBase {
+
+    @Override
+    public List<Event> visitVulkanRMW(VulkanRMW e) {
+        Register resultRegister = e.getResultRegister();
+        String mo = e.getMo();
+        Expression address = e.getAddress();
+        Register dummy = e.getFunction().newRegister(resultRegister.getType());
+        Load load = newRMWLoadWithMo(dummy, address, Tag.Vulkan.loadMO(mo));
+        RMWStore store = newRMWStoreWithMo(load, address, e.getValue(), Tag.Vulkan.storeMO(mo));
```

Comment:
If the RMW had `ATOM` tag, we need to assign this also to the load and the store. Maybe also other tags.

Reply:
Added `assignTag()` to add tags from RMW to load and store, but the `releaseseq2-filter.litmus` still fails, I need to do a further check

Reply:
You are missing the `ATOM` tag.

Also, `propagateTags` might be a better name and probably it should only take `(Event source, Event target)` as input and simply call it twice for the read and the write. It might be that we need to use this method further so I would go for a more general implemenation

Reply:
Function added, I also update `PTX` Tags to use it.

Reply:
I don't like this solutions for several reasons.
- You are creating Load and Store events without a mo, but "force-pushing" the mo tags. While this might work, it is not aligned with the way we do this for all other models, e.g. if you print the program, you will not see its mo as you should. You should keep the old `newRMWStoreWithMo(... Tag.PTX.storeMO(mo));`, etc
- Once you don't use these for propagating memory orders, old map are basically the identity, so it seems like an overkill.

You can probably have something as simple as (one for each Tag.Arch)
```
public static void propagateTags(Event source, Event target) {
        for (String tag : Set.of(CTA, GPU, SYS, GEN, TEX, SUR, CON)) {
            if (source.hasTag(tag)) {
                target.addTags(tag);
            }
        }
    }
```

Reply:
Updated, mine seems to be over complicated, this looks much cleaner.